### PR TITLE
CI: add a book build-deploy job

### PR DESCRIPTION
**Pull request by bilelmoussaoui**

### CODE REVIEW

File: .github/workflows/book.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          override: true
+
+      - run: cargo build
+        working-directory: book/listings
+
+  build-deploy:
+    runs-on: ubuntu-20.04
+    name: build
+    container:
+      image: ghcr.io/gtk-rs/gtk4-rs/gtk4:latest
```
Comment (Hofer-Julian) [inline .github/workflows/book.yml:32]: Why do I need gtk4 for deploying?

File: .github/workflows/book.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+          destination_dir: book/git
+
+  codespell:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: codespell-project/actions-codespell@master
+        with:
+          check_filenames: true
+          path: book
```
Comment (Hofer-Julian) [inline .github/workflows/book.yml:65]: I guess it wouldn't hurt to have codespell for the whole project?

Comment (bilelmoussaoui) [reply .github/workflows/book.yml:65]: Most of the code is auto-generated and will need a bunch of fixes which are unrelated to the book, maybe in the future

File: .github/workflows/book.yml (hunk 1 above)
Comment (bilelmoussaoui) [reply .github/workflows/book.yml]: Good point, removed

**Comment by bilelmoussaoui** [comment]
Let's give this a try and see if it works as expected o/

**Pull request merged**